File: models/events/client_events/clientevent.py
from pydantic import BaseModel
from enum import Enum

class ClientEventType (Enum):
    INTERACTION = "interaction"
    CLIENT_PAGE = "client_page"
    PONG = "pong"
    """A responed for ping events."""

class ClientEvent (BaseModel):
    sessionId: str
    event_time: int
    event_type: ClientEventType
    event_data: dict

File: web/js/models/clientevent.js


export const ClientEventType = Object.freeze({
    INTERACTION: "interaction"
});

export class ClientEvent {
    constructor (data = {}) {
        this.sessionId = data.sessionId;
        this.event_time = data.event_time;
        this.event_type = data.event_type;
        this.event_data = data.event_data;
    }
}